src/shogun/mathematics/RandomMixin.h

```diff
@@ -28,16 +29,17 @@ namespace shogun
 	private:
 		void init()
 		{
-			set_random_seed();
-
-			using this_t = RandomMixin<Parent, PRNG, RandomDevice>;
-
 			Parent::watch_param("seed", &m_seed);
-			Parent::template watch_method<bool>("seed_callback", [&]() {
-				m_prng = PRNG(m_seed);
-				seed_callback(this, m_seed);
-				return true;
-			});
+			Parent::template watch_method<bool>(
+			    "seed_callback",
+			    [&]() {
+				    m_prng = PRNG(m_seed);
+				    seed_callback(this, m_seed);
+				    return true;
+			    },
```

Comment:
You can try and make this void instead, I think Any supports void functions now. Does this compile though? It's a stateful lambda that is then converted to a function pointer..?